### PR TITLE
Bug/230 alarm rs

### PR DESCRIPTION
**Pull request by crbrox**
Raise/release alarm DATABASE/DATABASE_ORION when replica-set is used.

mongo driver has been updated, as the old one crashed when replica-set was down

### CODE REVIEW

File: config.js [hunk 1]
```diff
@@ -86,7 +86,7 @@ config.orionDb = {
 config.perseoCore = {
     rulesURL : 'http://localhost:8080/perseo-core/rules',
     noticesURL : 'http://localhost:8080/perseo-core/events',
-    interval: 60e3*1
+    interval: 60e3*5
```
Comment (crbrox) [inline config.js]: fixed ba4a71b

File: config.js [hunk 2]
```diff
@@ -182,8 +182,10 @@ config.DEFAULT_SERVICE= 'unknownt';
  * @type {Number}
  */
 config.checkDB = {
-    delay: 2000,
-    reportInterval: 15e3
+    delay: 5e3,
```
Comment (crbrox) [inline config.js]: changed ba4a71b

File: documentation/configuration.md [hunk 3]
```diff
@@ -47,7 +47,11 @@ In order to have perseo running, there are several basic pieces of information t
 * `config.mongo.URL`: URL for connecting mongoDB.
 * `config.executionsTTL`: Time-To-Live for documents of action executions (seconds).
 * `config.checkDB.delay`:  Number of milliseconds to check DB connection.
-* `config.checkDB.reportInterval`:  Number of milliseconds to report a problem with DB connection in logs.
+* `config.checkDB.reconnectTries`:  Number of of attempts to reconnect.
+* `config.checkDB.reconnectInterval`:  Number of milliseconds to wait between attempts to reconnect.
+* `config.checkDB.bufferMaxEntries`:  Number of operations buffered up before giving up on getting a working connection.
+
```
Comment (crbrox) [inline documentation/configuration.md]: firxed da50576

**Comment by fgalan** [comment]
LGTM

File: documentation/admin.md [hunk 4]
```diff
@@ -47,6 +47,43 @@ can be used to check the process is listening in the appropriate port (provided
...
+with DB.
+
+At runtime the connection to DB is managed by the driver. The driver keeps a
+buffer of operations waiting for connection. There is a limit for that buffer,
+established by the `checkDB.bufferMaxEntries` configuration parameter. Thus,
+if a connection problem persist, then the buffer size will eventually overpass
+the limit (all the operations waiting in the buffer will result in error in
+this case). In addition, an DB alarm at `ERROR` level is traced in the logs.
+
+Moreover, in standalone mode (and *not* in replica set modde) the driver
```
Comment (crbrox) [inline documentation/admin.md]: typo `mode`

File: documentation/admin.md [hunk 5]
```diff
@@ -47,6 +47,43 @@ can be used to check the process is listening in the appropriate port (provided
...
+also use a couple of parameters: `checkDB.reconnectTries` and
+`checkDB.reconnectInterval` to manage DB connections retries in the case of connection
+problems. If the connection to the server fails in this case, then the driver
+will try to reconnect as many time as `reconnectTries`, waiting `reconnectInterval`
```
Comment (crbrox) [inline documentation/admin.md]: `as many timeS`

**Comment by crbrox** [comment]
LGTM

**Pull request merged**